public/dorequest.php

```diff
@@ -147,11 +147,16 @@ function DoRequestError($errorMsg): void
         $emulatorId = requestInput('e');
         $consoleId = requestInput('c');
 
-        if ($emulatorId === null && $consoleId !== null) {
+        if (empty($emulatorId) && !empty($consoleId)) {
             DoRequestError("Lookup by Console ID has been deprecated");
             break;
         }
 
+        if (empty($emulatorId)) {
```

Comment:
shouldn't this logic be tied to the `integration_id` data in releases.php?

Reply:
I thought of it more like validation before passing it along but it might as well be handled in `getEmulatorReleaseByIntegrationId()`, yeah. If I understand correctly that is.

Reply:
oh, missed that next line. isn't this redundant then?

Reply:
as `getEmulatorReleaseByIntegrationId(int $integrationId)` requires an input it might be caught before even attempting to look it up. Admittedly, in this case it's not that helpful - it's just an array lookup - yet, we query the database a lot for e.g. empty usernames and rely on the query to not find anything instead of preventing even being run because the input was empty in the first place.